modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -1568,6 +1567,11 @@ private static DecimalFormat doubleFormat() {
     private GridProcessor securityProcessor() throws IgniteCheckedException {
         GridSecurityProcessor prc = createComponent(GridSecurityProcessor.class, ctx);
 
+        if (cfg.isAuthenticationEnabled() && !(prc instanceof IgniteAuthenticationProcessor)) {
```

Comment:
Why do we need this check?

Reply:
To prevent configuration of external security plugin and Ignite authentication at the same time.